### ISSUE TITLE
Not possible to retrieve child validator from ChildValidatorAdaptor anymore

### ISSUE THREAD

**Issue by bnordli** [JeremySkinner/FluentValidation#198]
With e1a80fed, the public Validator property was removed from ChildValidatorAdaptor, and the only way of retrieving the underlying validator is now the protected GetValidator(context) method. 

We rely on traversing child validators to dynamically generate client side validators, so this is a breaking change for us.

The work around is to use reflection to retrieve the child validator, but it should really have a public way of accessing the validator.

**Comment by JeremySkinner** [comment]
The child validator can now be instantiated lazily using a delegate, which is why the Validator property was removed. You can invoke this delegate to retrieve the validator instance using the GetValidator method- if you don't have a context available at this time then you can always supply a fake one with a null instance (assuming you're not relying on lazy instantiation of child validators):

```
var context = new ValidationContext(null);
adaptor.GetValidator(context);
```

**Comment by bnordli** [comment]
Thanks for the quick reply!

My problem is that the GetValidator method is _protected_, and not accessible without reflection.

(A lesser issue is that the GetValidator actually takes a concrete `PropertyValidatorContext` instance, so currently I actually have to do this in order to get to the validator:

```
var getter = typeof (ChildValidatorAdaptor).GetMethod("GetValidator", BindingFlags.NonPublic | BindingFlags.Instance);
var nullContext = new PropertyValidatorContext(new ValidationContext(null), null, null);
var childValidator = getter.Invoke(validator, new object[] {nullContext}) as IValidator;
```
)

**Comment by JeremySkinner** [comment]
Yes...I meant I can modify it and make it public. Sorry if that wasn't clear

**Comment by bnordli** [comment]
Yes, that would suffice for us! Great, thanks :)

**Comment by JeremySkinner** [comment]
Great, I'll try and get that done before the end of this week

**Comment by bnordli** [comment]
Thanks for the fix! :) 7f71912a680ef3aa
